data/scripts/dependency-resolver.py

```
@@ -144,40 +205,125 @@ def handle_python_check(args, conf):
+        return
+
+    supported = []
+    for i in check_cflags:
+        # must acumulate the tested one so we handle dependent flags like -Wformat*
+        flags = "%s %s" % (" ".join(supported), i)
+        success = compile_test(source, args.compiler, "-Werror %s" % flags, None)
+        if success:
+            supported.append(i)
+
+    if supported:
+        context.add_append_makefile_var(append_to, " ".join(supported))
 
 type_handlers = {
     "pkg-config": handle_pkgconfig_check,
     "ccode": handle_ccode_check,
     "exec": handle_exec_check,
     "python": handle_python_check,
+    "cflags": handle_cflags_check,
 }
 
-def var_str(items):
+def format_makefile_var(items):
     output = ""
     for k,v in items:
-        if not v: continue
-        output += "%s ?= %s\n" % (k, v)
+        if not v or not v["value"]: continue
+        output += "%s %s %s\n" % (k, v["attrib"], v["value"])
     return output
 
-def makefile_gen(args):
-    output = ""
-    output += var_str(makefile_vars.items())
-    output += var_str(cfg_cflags.items())
-    output += var_str(cfg_ldflags.items())
+def makefile_gen(args, context):
+    output = format_makefile_var(context.get_makefile_vars().items())
     f = open(args.makefile_output, "w+")
     f.write(output)
     f.close()
 
-def kconfig_gen(args):
+def kconfig_gen(args, context):
     output = ""
-    for k,v in cfg_kconfig.items():
-        output += "config {config}\n{indent}bool\n{indent}default {enabled}\n". \
-                  format(config=k, indent="       ", enabled=v)
+    for k,v in context.get_kconfig().items():
+        output += "config {config}\n{indent}{ktype}\n{indent}default {enabled}\n". \
+                  format(config=k, indent="       ", ktype=v["type"], enabled=v["value"])
     f = open(args.kconfig_output, "w+")
     f.write(output)
     f.close()
 
+def run(args, dep_checks, context):
+    for i in dep_checks:
+        handler = type_handlers.get(i["type"])
```

This will raise a `KeyNotFound` exception instead of returning `None`. Pass a `None` as the second parameter to the `get()` method.